plugins/ssr/ssrparser.mlg

```diff
@@ -178,27 +178,27 @@ let wit_ssrhoirep = add_genarg "ssrhoirep" (fun env sigma -> pr_hoi)
 
 let intern_ssrhoi ist = function
   | Hyp h -> Hyp (intern_hyp ist h)
-  | Id (SsrHyp (_, id)) as hyp ->
-    let _ = Tacintern.intern_genarg ist (in_gen (rawwit wit_ident) id) in
+  | Id (SsrHyp (loc, id)) as hyp ->
+    let _ = Tacintern.intern_genarg ist (in_gen (rawwit wit_identref) (CAst.make ?loc id)) in
     hyp
 
 let interp_ssrhoi ist gl = function
   | Hyp h -> let s, h' = interp_hyp ist gl h in s, Hyp h'
   | Id (SsrHyp (loc, id)) ->
-    let s, id' = interp_wit wit_ident ist gl id in
-    s, Id (SsrHyp (loc, id'))
+    let s, id' = interp_wit wit_identref ist gl (CAst.make ?loc id) in
+    s, Id (SsrHyp (loc, id'.CAst.v))
```

Comment:
I guess this diff hunk supports @ppedrot thesis. Here the loc is already in the AST, the copy in the ident is just getting in the way. It would feel more natural to pass loc to interp_wit frankly.

Reply:
I would say that it is expected that `interp_wit` takes a loc since it goes from user-facing syntax to "abstract" syntax (the first line of the diff). What is more questionable (and what @ppedrot questions, afaiu) is that `interp_wit` continues to return the loc (the second line of the diff).

Reply:
What I meant is that if you don't put the loc in the ident in the fist place, then interp_wit is not going to return it, you can still pass the loc as you pass ist and gl if you need to